### ISSUE TITLE
how to get last 4 layers in classification

### ISSUE THREAD

**Issue by radiodee1** [google-research/bert#582]
hi,

I'm using bert base for a classification problem. How do I get ahold of the last 4 layers of the model so that I can concat them? My classification problem presently inputs 768 and outputs 30,522 which I use to identify a single vocabulary word. I think my model would work better if I had the last four layers. Any help would be appreciated. Thanks!

**Comment by radiodee1** [comment]
Hi again. I'm trying something like this. This is my create-model function. I am trying here to get the last four layers. Would this do it? I also copy one of the DataProcessor classes. I can get the full vocab output by modifying the 'get_labels' function. Is this how it would be done by someone else? Thanks. LAYERS is an array like this: 

    LAYERS = [-1, -2, -3, -4]

Here is the long listing:


    def create_model(bert_config, is_training, input_ids, input_mask, segment_ids,
                     labels, num_labels, use_one_hot_embeddings, unique_ids=None):
        """Creates a classification model."""
        global model

        if True:
            model = modeling.BertModel(
                config=bert_config,
                is_training=is_training,
                input_ids=input_ids,
                input_mask=input_mask,
                token_type_ids=segment_ids,
                use_one_hot_embeddings=use_one_hot_embeddings)

        # In the demo, we are doing a simple classification task on the entire
        # segment.
        #
        # If you want to use the token-level output, use model.get_sequence_output()
        # instead.
        #output_layer = model.get_sequence_output() ## ??

        #i = 0
        layers = []
        layer_indexes = LAYERS
        all_layers = model.get_all_encoder_layers()

        if True:
            for (j, layer_index) in enumerate(layer_indexes):
                layer_output = all_layers[layer_index]

                layer_output_flat = tf.map_fn(lambda x: x, layer_output)
                layer_output_flat = tf.squeeze(layer_output_flat[:, -1:, :], axis=1) ## choose the last!!
                layers.append(layer_output_flat)

            output_layer = tf.concat(layers, 1)
            print(output_layer,':out')

        else:
            output_layer = model.get_pooled_output() ## <-- remove

        hidden_size = output_layer.shape[-1].value

        output_weights = tf.get_variable(
            "output_weights", [num_labels, hidden_size],
            initializer=tf.truncated_normal_initializer(stddev=0.02))

        output_bias = tf.get_variable(
            "output_bias", [num_labels], initializer=tf.zeros_initializer())

        with tf.variable_scope("loss"):
            if is_training:
                # I.e., 0.1 dropout
                output_layer = tf.nn.dropout(output_layer, keep_prob=0.9)

            logits = tf.matmul(output_layer, output_weights, transpose_b=True)
            logits = tf.nn.bias_add(logits, output_bias)
            probabilities = tf.nn.softmax(logits, axis=-1)
            log_probs = tf.nn.log_softmax(logits, axis=-1)

            one_hot_labels = tf.one_hot(labels, depth=num_labels, dtype=tf.float32)

            per_example_loss = -tf.reduce_sum(one_hot_labels * log_probs, axis=-1)
            loss = tf.reduce_mean(per_example_loss)

            return (loss, per_example_loss, logits, probabilities)

**Comment by SlimenBouras** [comment]
Hi , did you get the solutions please ?

**Comment by radiodee1** [comment]
i couldn't figure it out
